Game.cs

```diff
@@ -2,28 +2,73 @@
 using Godot.Collections;
 using JetBrains.Annotations;
 
-[UsedImplicitly]
-public class Game : Node
+[UsedImplicitly] public class Game : Node
 {
     private AudioStreamPlayer musicPlayer;
     private Inventory inventory;
-    private Map map;
+    [CanBeNull] private Map map;
+    private Viewport viewport;
+    private Transition transition;
+
+    private readonly PackedScene mapScene = GD.Load<PackedScene>("res://tilemap/Map.tscn");
+    private readonly PackedScene gameOverScene = GD.Load<PackedScene>("res://game-over/GameOver.tscn");
 
     public override void _Ready()
     {
         musicPlayer = GetNode<AudioStreamPlayer>("AudioStreamPlayer");
         inventory = GetNode<Inventory>("UI/Inventory");
-        map = GetNode<Map>("ViewportContainer/Viewport/Map");
-        for (int i = 0; i < 4; i++)
+        viewport = GetNode<Viewport>("ViewportContainer/Viewport");
+        transition = GetNode<Transition>("Transition");
+
+        inventory.Connect(nameof(Inventory.HeartsRanOut), this, nameof(OnGameOver));
+        StartGame();
+    }
+
+    private async void Retry(GameOver gameOver)
+    {
+        gameOver.Disconnect(nameof(GameOver.OnRetry), this, nameof(Retry));
+        var img = viewport.GetTexture().GetData();
+        transition.Start(img);
+        if (map != null)
+        {
+            map?.QueueFree();
+            map = null;
+        }
+
+        // await ToSignal(transition, nameof(Transition.TransitionMidPoint));
+        await ToSignal(transition, nameof(Transition.TransitionEnd));
+        StartGame();
+    }
```

Comment:
This happens when the player presses the H button to retry (ideally any button), I don't think it works very well :sweat_smile: